### ISSUE TITLE
Mutable Query object in v6

### ISSUE THREAD

**Issue by ogtfaber** [invertase/react-native-firebase#2691]
## Issue

Query objects seem to be mutable now.

```
const query = db.collection('users').where('age', '>', 30)
const query2 = query.orderBy('updatedAt')
```

The Query object `query` now also has the order clause included.

**Comment by Ehesp** [comment]
Please follow the issue template. This looks like an issue, however we need to know what version you are on.

**Comment by Salakar** [comment]
@ogtfaber thanks for reporting this, I've pushed up a fix for this (plus new tests) in Firestore & also RTDB as it seems both were mutating previous query instances.

**Firestore**: https://github.com/invertase/react-native-firebase/commit/8dc6e830c086c9c8d22691a18e1d4019b1370d67
**Database**: https://github.com/invertase/react-native-firebase/commit/147ee635bd6d657af6c39b2fc9209a9afa380830

This will be in the patch release later today.

**Comment by Salakar** [comment]
Hey 👋 this issue should now be fixed in v6.0.1, thanks for the report. [[release notes]](https://invertase.io/oss/react-native-firebase/releases/v6.0.1)

**Comment by ogtfaber** [comment]
Thank you! @Salakar
